options.go

```diff
@@ -84,6 +87,12 @@ func WithCredentialsJSON(b []byte) Option {
 			d.err = errtype.NewConfigError(err.Error(), "n/a")
 			return
 		}
+		ud, err := c.GetUniverseDomain()
+		if err != nil {
+			d.err = errtype.NewConfigError(err.Error(), "n/a")
+			return
+		}
+		d.authUniverseDomain = ud
```

Comment:
Nit: `credentialsUniverse` reads better to me, since "auth" is overloads with Auto IAM AuthN, database authn, and ADC auth.